### PR TITLE
[develop] Removing RRFS-related features.

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
## DESCRIPTION OF CHANGES: 

After recent high-level decisions to discontinue support of GSI, we have decided to discontinue support for any cycling with GSI in SRW. This PR aims to remove all the unnecessary components that have been added to the App consistent with those decisions.

Specifically, the following changes were made:

* The external repositories for GSI and rrfs-utils were removed from Externals.cfg.
* The build script options and CMake options for both GSI and rrfs-utils were removed.
* The j-jobs and ex-scripts related to cycling were removed.
* The parm/wflow settings for DA data and cycling were removed, along with any config files in ush and the test directories.
* The GSI-required libraries were removed from the module files.
* DA Obs entries were removed from the data locations file.
* RRFS concepts related to cycling and restarting were removed from configuration options and scripts.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 

- [x] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [x] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)
It would be helpful if someone could test this branch to ensure it doesn't cause problems for AQM. I have not tested that capability.

## DEPENDENCIES:

None.

## DOCUMENTATION:

@gspetro-NOAA Let me know what makes sense in terms of updating docs.

## ISSUE: 

None.

## CHECKLIST

- [ ] My code follows the style guidelines in the Contributor's Guide
- [ ] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [ ] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Review by MichaelLueken — CHANGES_REQUESTED** [review]
@christinaholtNOAA - Thank you so much for the quick turnaround on removing RRFS-related features from the SRW App!  The overall, the changes look good to me.  However, it looks like two `modulefiles/tasks/` files should also be removed: [get_da_obs.local.lua](https://github.com/christinaholtNOAA/ufs-srweather-app/blob/remove_gsi/modulefiles/tasks/hera/get_da_obs.local.lua) on Hera and [get_da_obs.local.lua](https://github.com/christinaholtNOAA/ufs-srweather-app/blob/remove_gsi/modulefiles/tasks/jet/get_da_obs.local.lua) on Jet.

I've also noted that the `grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot` fundamental WE2E test is failing on Hera.  Did you encounter this during your testing on Hera?  The `plot_allvars` task is failing with the following:

```
Traceback (most recent call last):
  File "/scratch2/NAGAPE/epic/Michael.Lueken/ufs-srweather-app/scripts/exregional_plot_allvars.py", line 363, in <module>
    data1 = pygrib.open(
  File "src/pygrib/_pygrib.pyx", line 323, in pygrib._pygrib.open.__cinit__
OSError: [Errno could not open %s] /scratch2/NAGAPE/epic/Michael.Lueken/expt_dirs/grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot/2019070100//postprd/rrfs.t00z.prslev.f000.rrfs_conus_25km.grib2
```
The issue is that the script is looking for `rrfs.t00z.prslev.f000.rrfs_conus_25km.grib2`, but the correct name is now `srw.t00z.prslev.f000.rrfs_conus_25km.grib2`.  In `scripts/exregional_plot_allvars.py`, line 365, the `rrfs.t` is hardcoded in.  This should be changed to use `NET`, if possible.  Please note that this is also an issue with `scripts/exregional_plot_allvars_diff.py` as well.

The SRW App was built for AQM and the `aqm_grid_AQM_NA13km_suite_GFS_v16` WE2E test was run on Hera.  The test successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
aqm_grid_AQM_NA13km_suite_GFS_v16                                  COMPLETE            1005.66
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE            1005.66
```

**Comment by EdwardSnyder-NOAA** [comment]
From what I can recall, this should be all the rrfs changes. I ran a few WE2E tests on Hera and didn't have any troubles with it. I'll approve the PR knowing that @MichaelLueken changes will be addressed.

**Review by EdwardSnyder-NOAA — APPROVED** [review]

File: ush/config_defaults.yaml
Comment (mkavulich) [inline ush/config_defaults.yaml:8]: I feel as if this variable could still be valuable for tracking compatibility of config files. Especially for cases like this, where previously valid variables are removed. But it's fine to remove now I guess if it isn't currently used.

File: scripts/exregional_run_prdgen_subpiece.sh
Comment (mkavulich) [inline scripts/exregional_run_prdgen_subpiece.sh:46]: Same here, is this file still needed?

File: parm/data_locations.yml
Comment (mkavulich) [inline parm/data_locations.yml:325]: This is missing the `archive_file_names:` entry from the old `NSSL_mrms` stanza above.

File: scripts/exregional_run_prdgen.sh
Comment (mkavulich) [inline scripts/exregional_run_prdgen.sh:186]: Shouldn't this file be removed? I'm not sure what it's used for if not RRFS.

**Review by mkavulich — CHANGES_REQUESTED** [review]
One required change and a few comments/questions. You will also need to add the following lines to the tests `config.MET_ensemble_verification_only_vx.yaml` and `config.MET_verification_only_vx.yaml` to ensure they pass:

```
nco:
  NET_default: 'rrfs'
```

File: ush/config_defaults.yaml
Comment (christinaholtNOAA) [reply ush/config_defaults.yaml:8]: It isn't used, as far as I can tell. If it were, it would need to change with every change of this file. It would likely be best to just track the hash of the repo that was used for an experiment. That would be the only thing that would really come close to ensuring compatibility of config files, although I'm not sure even that would be a strong 100% guarantee.

Until there is some sort of automated method for updating this number, it's fallible.

File: parm/data_locations.yml
Comment (christinaholtNOAA) [reply parm/data_locations.yml:325]: Ooh. Good catch. Thanks.

File: scripts/exregional_run_prdgen.sh
Comment (christinaholtNOAA) [reply scripts/exregional_run_prdgen.sh:186]: I think this one and the one below are still needed/useful. This is just about creating subdomains from larger ones. They do happen to be the RRFS domains, but we were suggesting that we'd still support that part of SRW -- domains and physics.

**Review by christinaholtNOAA — COMMENTED** [review]
@mkavulich Thanks for the review. I've addressed a bunch of your changes locally, and have made a few others @gspetro-NOAA found while updating the docs. I will test those changes and then push.

**Comment by christinaholtNOAA** [comment]
I've pushed my tested changes here, and am running a final round of the fundamental tests after I merged `develop` into my aging branch. I'll report back on those results shortly.

When I ran the comprehensive test suite on hera before I merged `develop` my only failure was in the GST_release_public_v1 test. I observed the same failure as documented in Issue #874.

**Review by mkavulich — APPROVED** [review]
Thanks for addressing my comments!

**Review by natalie-perlin — APPROVED** [review]

**Comment by natalie-perlin** [comment]
Could Daniel Abdi's #540  be closed if RRFS-related features are removed at this point?
https://github.com/ufs-community/ufs-srweather-app/pull/540

**Review by MichaelLueken — APPROVED** [review]
@christinaholtNOAA - Thanks for addressing my concerns!  All of the fundamental tests are successfully passing now:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta    COMPLETE               8.31
nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_  COMPLETE              11.15
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2        COMPLETE               6.43
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v17_p8_plot  COMPLETE              12.51
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_HRRR_suite_HRRR          COMPLETE              25.76
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0              COMPLETE              13.20
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16                COMPLETE              18.68
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE              96.04
```
Approving now.

**Comment by christinaholtNOAA** [comment]
@MichaelLueken Thanks for the follow up. I am seeing the same on Hera when I checked back in on my latest run this morning.

**Review by MichaelLueken — CHANGES_REQUESTED** [review]
@christinaholtNOAA - Unfortunately, the Jet coverage tests failed because the `process_obs` WE2E test is still present in `tests/WE2E/machine_suites/coverage.jet`.  Please remove this test, then I can queue up the automated tests on Jet.  Thanks!

**Comment by MichaelLueken** [comment]
@christinaholtNOAA - It looks like there are also GSI related changes that should be removed in `modulefiles/build_gaea_intel.lua`. It looks like lines [19-24](https://github.com/ufs-community/ufs-srweather-app/blob/98c4106a3505191017094e7e51cf410243163d0b/modulefiles/build_gaea_intel.lua#L19-L24)  can be removed.  Looking at the changes that @natalie-perlin has made in PR #889, these lines have been removed.

@natalie-perlin - is it safe for @christinaholtNOAA to remove these lines as part of her removal of RRFS-related features in this PR?  Thanks!

**Review by MichaelLueken — APPROVED** [review]
Thank you very much, @christinaholtNOAA!  Reapproving now and resubmitting Gaea and Jet tests.

File: modulefiles/build_gaea_intel.lua
Comment (natalie-perlin) [inline modulefiles/build_gaea_intel.lua]: @christinaholtNOAA , 
all these could be removed, too, as they were needed for GSI only:
```
prepend_path("LD_LIBRARY_PATH",pathJoin(MKLROOT,"lib/intel64"))
pushenv("MKLROOT", MKLROOT)
pushenv("CRAYPE_LINK_TYPE","dynamic")
```

**Comment by MichaelLueken** [comment]
@christinaholtNOAA - The rerun of the Jenkins tests on Gaea successfully passed.  The Jet tests that require pulling data from HPSS failed to pull data from HPSS with the following errors:
```
ERROR: htar_BlockRead: Error 5 reading 288 bytes from HPSS-resident file: 
ERROR: htar_ReadIndexHeader: Error -1 reading HPSS-resident index file header: 
HTAR: HTAR FAILED
```
Resubmitting the tests this morning to see if they will pass.  Once successful, I can merge this work.

File: modulefiles/build_gaea_intel.lua
Comment (natalie-perlin) [reply modulefiles/build_gaea_intel.lua]: Alternatively, I could take care of these when updating the modulefiles #889 , as it may be scheduled to be merged after the present one (993)

Comment (christinaholtNOAA) [reply modulefiles/build_gaea_intel.lua]: @natalie-perlin I can go ahead and remove those here.

**Comment by MichaelLueken** [comment]
Just as a head's up, according to the Jet helpdesk, the issue with accessing Jet via the Princeton bastion is also causing issues with Jet accessing HPSS.

**Comment by MichaelLueken** [comment]
The Jenkins WE2E tests that previously failed on Jet, due to access issues to HPSS, have successfully passed.  Will now move forward with merging this PR.

**Pull request merged**
